app/controllers/sessions_controller.rb

```diff
@@ -7,7 +7,14 @@ def login
       session[:user_id] = @user.id
       @user.loginAttempts = 0
       @user.save
-      redirect_to homepage_path
+      # Checks to see if the password is close to expiring (30 days)
+      if remaining_days <= 0
+        redirect_to homepage_path, warning: "Your password is expired. Please change your password now."
```

Comment:
If the user's password expired why are you sending them to the `homepage_path`? Wouldnt you want to send them to `welcome_path`?

Reply:
Even better change it to `logout_path` and it will log them out and kick them to the welcome page. <-- That is how we should do it

Reply:
Okay, so I've adjusted it so that the password is checked before the user is signed in. If the password is expired, the user is not signed in and is sent back to the welcome page (no logout required). Otherwise, the session logs the user in, zeroes the login attempts, saves the user, and redirects the user to the homepage. If the password is within the 3 day expiration date, a notification prompting to change the password will appear.

Reply:
Nice. Yeah, that works fine I just don't like having duplicate code. Like how `session[:user_id] = @user.id` and resetting the attempts is used multiple times. It's fine for now but this whole login function needs to be redone. It's confusing and can definitely be cleaned up and simplified. (We should also make tests for it because that will make redoing it easer to check if its working)

Reply:
Added another if statement, so hopefully that resolves this card. I separated the session so it's not in two if statements but rather one.